client/client-lib/src/lib.rs

```diff
@@ -270,13 +270,13 @@ impl<T: AsRef<ClientConfig> + Clone> Client<T> {
             .context
             .api
             .fetch_peg_out_fees(&recipient, &amount)
-            .await?;
-        fees.map(|fees| PegOut {
+            .await
+            .map_err(|_| ClientError::PegOutWaitingForUTXOs)?;
```

Comment:
Isn't this only one failure case of many? What if the client is just offline? I think you need to distinguish between this specific error and general HTTP errors. Do we already have error codes for this kind of stuff?

Reply:
Right don't know what I was thinking there..

Reply:
Could we just define negative integers for specific API errors ? Right now we have 404, 400 but I think everything in the positive range is already defined for/by HTTP ?